### PR TITLE
Support faraday 2.0 & streaming api

### PR DESCRIPTION
**Pull request by aka-nez**
Hi! I tried to update the gem to make it compatible with Faraday 2 and add streaming API support.

Take a look. please @iMacTia 

CleanShot 2022-10-21 at 11 35 54@2x

Probably, I should update the CI matrix, too: remove 2.5, add 3.0+

### CODE REVIEW

**Review by iMacTia — COMMENTED** [review]
Thank you @aka-nez 🙌!

Agree about the CI matrix, please make it cover 2.6 -> 3.1 like we do in Faraday.

Out of curiosity, could you explain why you needed to explicitly add the `faraday/multipart` middleware? What kind of errors were you seeing otherwise?

**Comment by aka-nez** [comment]
> Out of curiosity, could you explain why you needed to explicitly add the `faraday/multipart` middleware? What kind of errors were you seeing otherwise?

spec/faraday/http/adapter_spec.rb:26 uses this middleware, that was bundled with Faraday before

**Comment by iMacTia** [comment]
Thanks for clarifying!
I saw you added it to the Gemfile and `spec_helper`, so there should be no impact on users of this gem 👍

**Comment by aka-nez** [comment]
I have updated the CI matrix and entirely removed everything related to multipart. It doesn't impact tests

**Comment by olleolleolle** [comment]
(I did some cleaning out of earlier Required Build Checks, which included now-removed ones.)

**Comment by olleolleolle** [comment]
Ah, I noted this in the failure in CI:

>   Gems in the groups 'development' and 'test' were not installed.

Perhaps some --deployment flag is enabled>

**Comment by aka-nez** [comment]
Removed this line from ci

File: faraday-http.gemspec [hunk 1]
```diff
@@ -24,12 +24,13 @@ Gem::Specification.new do |spec|
...
-  spec.add_dependency 'faraday', '~> 1.0'
+  spec.add_dependency 'faraday', '~> 2.0'
   spec.add_dependency 'http', '>= 4.0', '< 6'
 
   spec.add_development_dependency 'bundler', '~> 2.0'
   spec.add_development_dependency 'rake', '~> 13.0'
   spec.add_development_dependency 'rspec', '~> 3.0'
   spec.add_development_dependency 'simplecov'
   spec.add_development_dependency 'webmock', '~> 3.4'
+  spec.metadata['rubygems_mfa_required'] = 'true'
```
Comment (olleolleolle) [inline faraday-http.gemspec:35]: Suggestion: Can we move this line to sit just after "changelog_uri"?

Comment (aka-nez) [reply faraday-http.gemspec]: Sure. My bad, it was rubocop -a correction. Fixed.

**Comment by olleolleolle** [comment]
@iMacTia Are we ready to go, here?

**Review by iMacTia — CHANGES_REQUESTED** [review]
Actually, `faraday ~> 2.0` is incorrect, we should pin to `faraday ~> 2.5` as older versions don't have the [new streaming API](https://github.com/lostisland/faraday/releases/tag/v2.5.0). Tests are passing because bundler installs the latest version, which works fine.

**Comment by aka-nez** [comment]
> Actually, `faraday ~> 2.0` is incorrect, we should pin to `faraday ~> 2.5` as older versions don't have the [new streaming API](https://github.com/lostisland/faraday/releases/tag/v2.5.0). Tests are passing because bundler installs the latest version, which works fine.

Sure. Fixed! 👍

**Review by iMacTia — APPROVED** [review]

**Comment by iMacTia** [comment]
Thank you so much @aka-nez, I'm gonna drop a release soon 🙌 !

**Pull request merged**